src/backend/catalog/test/storage_pending_deletes_redo_test.c

```diff
@@ -0,0 +1,1139 @@
+		.relnode.node.relNode = TEST_REL_OID1
+	};
+
+	ls_transactions_comlpete = lappend_int(ls_transactions_comlpete, pd.xid);
+
+	PdlRedoAdd(&pd);
+
+	PdlRedoDropFiles();
+
+	assert_int_equal(DropRelationFiles_call_count, 0);
+
+	list_free(ls_transactions_comlpete);
+	ls_transactions_comlpete = NIL;
+}
+
+/*
+ * Scenario:
+ * add single pending delete node
+ * and remove pending deletes for that node's xid
+ * and then drop files.
+ */
+static void
+test_7(void **state)
+{
+	setup(7);
+	ShmemVariableCache->oldestXid = (TransactionId) 1;
+
+	PendingRelXactDelete pd =
+	{
+		.xid = (TransactionId) 1,
+		.relnode.isTempRelation = false,
+		.relnode.node.spcNode = TEST_TABLESPACE_OID1,
+		.relnode.node.dbNode = TEST_DB_OID1,
+		.relnode.node.relNode = TEST_REL_OID1
+	};
+
+	PdlRedoAdd(&pd);
+
+	PdlRedoRemoveTree(pd.xid, NULL, 0);
+
+	PdlRedoDropFiles();
+
+	assert_int_equal(DropRelationFiles_call_count, 0);
+}
+
+/*
+ * Scenario:
+ * add single pending delete node
+ * and remove pending deletes for different xid
+ * and then drop files.
+ */
+static void
+test_8(void **state)
+{
+	setup(8);
+	ShmemVariableCache->oldestXid = (TransactionId) 1;
+
+	PendingRelXactDelete pd =
+	{
+		.xid = (TransactionId) 1,
+		.relnode.isTempRelation = false,
+		.relnode.node.spcNode = TEST_TABLESPACE_OID1,
+		.relnode.node.dbNode = TEST_DB_OID1,
+		.relnode.node.relNode = TEST_REL_OID1
+	};
+
+	PdlRedoAdd(&pd);
+
+	PdlRedoRemoveTree(pd.xid + 1, NULL, 0);
+
+	PdlRedoDropFiles();
+
+	assert_int_equal(DropRelationFiles_call_count, 1);
+}
+
+/*
+ * Scenario:
+ * add single pending delete node
+ * and remove pending deletes for invalid xid
+ * and then drop files.
+ */
+static void
+test_9(void **state)
+{
+	setup(9);
+	ShmemVariableCache->oldestXid = (TransactionId) 1;
+
+	PendingRelXactDelete pd =
+	{
+		.xid = (TransactionId) 1,
+		.relnode.isTempRelation = false,
+		.relnode.node.spcNode = TEST_TABLESPACE_OID1,
+		.relnode.node.dbNode = TEST_DB_OID1,
+		.relnode.node.relNode = TEST_REL_OID1
+	};
+
+	PdlRedoAdd(&pd);
+
+	PdlRedoRemoveTree(InvalidTransactionId, NULL, 0);
+
+	PdlRedoDropFiles();
+
+	assert_int_equal(DropRelationFiles_call_count, 1);
+}
+
+/*
+ * Scenario:
+ * add several pending delete node with the same xid
+ * and remove pending deletes for that xid
+ * and then drop files.
+ */
+static void
+test_10(void **state)
+{
+	setup(10);
+	ShmemVariableCache->oldestXid = (TransactionId) 1;
+
+	PendingRelXactDelete pd =
+	{
+		.xid = (TransactionId) 1,
+		.relnode.isTempRelation = false,
+		.relnode.node.spcNode = TEST_TABLESPACE_OID1,
+		.relnode.node.dbNode = TEST_DB_OID1,
+		.relnode.node.relNode = TEST_REL_OID1
+	};
+
+	PdlRedoAdd(&pd);
+
+	pd.relnode.node.spcNode = TEST_TABLESPACE_OID1;
+	pd.relnode.node.dbNode = TEST_DB_OID1;
+	pd.relnode.node.relNode = TEST_REL_OID2;
+
+	PdlRedoAdd(&pd);
+
+	pd.relnode.node.spcNode = TEST_TABLESPACE_OID1;
+	pd.relnode.node.dbNode = TEST_DB_OID2;
+	pd.relnode.node.relNode = TEST_REL_OID1;
```

Comment:
I suggest to rewrite this code to make it easier to understand what is changed
```suggestion
	pd.relnode.node.relNode = TEST_REL_OID2;

	PdlRedoAdd(&pd);

	pd.relnode.node.dbNode = TEST_DB_OID2;
	pd.relnode.node.relNode = TEST_REL_OID1;
```